### ISSUE TITLE
upgrade failing on with [schema_registry - Tried to build a global schema for view ks1.table1_col6_idx_index with an uninitialized base]

### ISSUE THREAD

**Issue by fruch** [scylladb/scylladb#15736]
## Issue description

- [x] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

when upgrading node-3, when it start with the new version it crash like this, multiple times:
```
Oct 14 06:41:15 rolling-upgrade--centos-db-node-e9674e0f-0-3 scylla[24027]:  [shard 0:stre] schema_registry - Tried to build a global schema for view ks1.table1_col6_idx_index with an uninitialized base info, at: 0x5fa1cbe 0x5fa2280 0x5fa2558 0x5a73dd7 0x1fbe73b 0x30fad22 0x32fdfef 0x139205a 0x5aa4edf 0x5aa61b7 0x5aa5529 0x5a47e07 0x5a46fbc 0x1315b0f 0x1317580 0x131405c /opt/scylladb/libreloc/libc.so.6+0x27b89 /opt/scylladb/libreloc/libc.so.6+0x27c4a 0x1311ca4

2023-10-14 06:48:56.144 <2023-10-14 06:48:56.117>: (DatabaseLogEvent Severity.ERROR) period_type=one-time event_id=383f8bdd-4b30-47b9-a436-d07b3bd98d9c: type=ABORTING_ON_SHARD regex=Aborting on shard line_number=29996 node=rolling-upgrade--centos-db-node-e9674e0f-0-3
2023-10-14T06:48:56.117+00:00 rolling-upgrade--centos-db-node-e9674e0f-0-3     !INFO | scylla[25417]: Aborting on shard 2.
?? ??:0
?? ??:0
?? ??:0
?? ??:0
?? ??:0
?? ??:0
HUF_decompress4X2_usingDTable_internal_fast_c_loop at short_streams.cc:?
seastar::future_state_base::any::available() const at ././seastar/include/seastar/core/future.hh:432
(inlined by) seastar::future_state_base::available() const at ././seastar/include/seastar/core/future.hh:509
(inlined by) seastar::future<mutation_fragment_v2*>::wait() at ././seastar/include/seastar/core/future.hh:1344
(inlined by) seastar::future<mutation_fragment_v2*>::get() at ././seastar/include/seastar/core/future.hh:1316
(inlined by) seastar::future<mutation_fragment_v2*>::get0() at ././seastar/include/seastar/core/future.hh:1335
(inlined by) seastar::internal::awaiter<true, mutation_fragment_v2*>::await_resume() at ././seastar/include/seastar/core/coroutine.hh:178
(inlined by) evictable_reader_v2::fill_buffer() at ./readers/multishard.cc:616
cql3::query_processor::query_processor(service::storage_proxy&, service::forward_service&, data_dictionary::database, service::migration_notifier&, service::migration_manager&, cql3::query_processor::memory_config, cql3::cql_config&, utils::loading_cache_config, service::raft_group0_client&, std::optional<wasm::startup_context>) at query_processor.cc:?
~vector at /usr/bin/../lib/gcc/x86_64-redhat-linux/12/../../../../include/c++/12/bits/stl_vector.h:730
(inlined by) ~token_range_endpoints at ././dht/token_range_endpoints.hh:22
(inlined by) void std::destroy_at<dht::token_range_endpoints>(dht::token_range_endpoints*) at /usr/bin/../lib/gcc/x86_64-redhat-linux/12/../../../../include/c++/12/bits/stl_construct.h:88
(inlined by) void std::_Destroy<dht::token_range_endpoints>(dht::token_range_endpoints*) at /usr/bin/../lib/gcc/x86_64-redhat-linux/12/../../../../include/c++/12/bits/stl_construct.h:149
(inlined by) void std::_Destroy_aux<false>::__destroy<dht::token_range_endpoints*>(dht::token_range_endpoints*, dht::token_range_endpoints*) at /usr/bin/../lib/gcc/x86_64-redhat-linux/12/../../../../include/c++/12/bits/stl_construct.h:163
(inlined by) void std::_Destroy<dht::token_range_endpoints*>(dht::token_range_endpoints*, dht::token_range_endpoints*) at /usr/bin/../lib/gcc/x86_64-redhat-linux/12/../../../../include/c++/12/bits/stl_construct.h:195
(inlined by) void std::_Destroy<dht::token_range_endpoints*, dht::token_range_endpoints>(dht::token_range_endpoints*, dht::token_range_endpoints*, std::allocator<dht::token_range_endpoints>&) at /usr/bin/../lib/gcc/x86_64-redhat-linux/12/../../../../include/c++/12/bits/alloc_traits.h:850
(inlined by) ~vector at /usr/bin/../lib/gcc/x86_64-redhat-linux/12/../../../../include/c++/12/bits/stl_vector.h:730
(inlined by) db::token_ring_table::execute(reader_permit, db::result_collector&, db::virtual_table::query_restrictions const&) at ./db/system_keyspace.cc:2061
void seastar::logger::log<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, seastar::tasktrace>(seastar::log_level, seastar::logger::format_info, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, seastar::tasktrace&&) at ././seastar/include/seastar/util/log.hh:234
(inlined by) void seastar::logger::error<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, seastar::tasktrace>(seastar::logger::format_info, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, seastar::tasktrace&&) at ././seastar/include/seastar/util/log.hh:328
(inlined by) operator() at ./main.cc:499
(inlined by) ~deferred_action at ././seastar/include/seastar/util/defer.hh:63
(inlined by) ~shared_ptr_count_for at ././seastar/include/seastar/core/shared_ptr.hh:481
?? ??:0
?? ??:0
?? ??:0
HUF_decompress1X1_usingDTable_internal_default at short_streams.cc:?
?? ??:0
?? ??:0
```

## Impact

user can't finish upgrading the cluster

## How frequently does it reproduce?

seen on multiple case of rolling upgrades, will update on comments

## Installation details

Kernel Version: 3.10.0-1160.99.1.el7.x86_64
Scylla version (or git commit hash): `5.3.0~rc0-20230523.3c3621db072d` with build-id `78cb9caaf1cdd0b37c3a38fa83548e6c4e3d4ab4`
Target Scylla version (or git commit hash): `5.4.0~dev-20231006.498e3ec435be` with build-id `16c6112202348a8adba536b4195d48adfdf958f9`

Cluster size: 4 nodes (n2-highmem-8)

Scylla Nodes used in this run:
- rolling-upgrade--centos-db-node-e9674e0f-0-4 (35.231.101.254 | 10.142.15.200) (shards: 7)
- rolling-upgrade--centos-db-node-e9674e0f-0-3 (34.148.205.28 | 10.142.15.196) (shards: 7)
- rolling-upgrade--centos-db-node-e9674e0f-0-2 (34.139.244.171 | 10.142.15.194) (shards: 7)
- rolling-upgrade--centos-db-node-e9674e0f-0-1 (35.237.33.46 | 10.142.0.123) (shards: 7)

OS / Image: `https://www.googleapis.com/compute/v1/projects/centos-cloud/global/images/family/centos-7` (gce: undefined_region)

Test: `rolling-upgrade-centos7-test`
Test id: `e9674e0f-a1df-4cd0-9125-3c510e7de33e`
Test name: `scylla-master/rolling-upgrade/rolling-upgrade-centos7-test`
Test config file(s):

- [rolling-upgrade.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/9d973065b179412f57dd2f493ddca78b644b76d0/test-cases/upgrades/rolling-upgrade.yaml)

<details>
<summary>
Logs and commands
</summary>

- Restore Monitor Stack command: `$ hydra investigate show-monitor e9674e0f-a1df-4cd0-9125-3c510e7de33e`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=e9674e0f-a1df-4cd0-9125-3c510e7de33e)
- Show all stored logs command: `$ hydra investigate show-logs e9674e0f-a1df-4cd0-9125-3c510e7de33e`

## Logs:
- **db-cluster-e9674e0f.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e9674e0f-a1df-4cd0-9125-3c510e7de33e/20231014_065350/db-cluster-e9674e0f.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e9674e0f-a1df-4cd0-9125-3c510e7de33e/20231014_065350/db-cluster-e9674e0f.tar.gz)
- **sct-runner-events-e9674e0f.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e9674e0f-a1df-4cd0-9125-3c510e7de33e/20231014_065350/sct-runner-events-e9674e0f.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e9674e0f-a1df-4cd0-9125-3c510e7de33e/20231014_065350/sct-runner-events-e9674e0f.tar.gz)
- **sct-e9674e0f.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e9674e0f-a1df-4cd0-9125-3c510e7de33e/20231014_065350/sct-e9674e0f.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e9674e0f-a1df-4cd0-9125-3c510e7de33e/20231014_065350/sct-e9674e0f.log.tar.gz)
- **loader-set-e9674e0f.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e9674e0f-a1df-4cd0-9125-3c510e7de33e/20231014_065350/loader-set-e9674e0f.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e9674e0f-a1df-4cd0-9125-3c510e7de33e/20231014_065350/loader-set-e9674e0f.tar.gz)
- **monitor-set-e9674e0f.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e9674e0f-a1df-4cd0-9125-3c510e7de33e/20231014_065350/monitor-set-e9674e0f.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e9674e0f-a1df-4cd0-9125-3c510e7de33e/20231014_065350/monitor-set-e9674e0f.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/rolling-upgrade/job/rolling-upgrade-centos7-test/315/)
[Argus](https://argus.scylladb.com/test/3f60c813-ab37-4cea-b400-21e8d4fc6dc9/runs?additionalRuns[]=e9674e0f-a1df-4cd0-9125-3c510e7de33e)
</details>

**Comment by fruch** [comment]
https://github.com/scylladb/scylladb/issues/14011 was opened on May,

this has started failing/coredumping half of our rolling upgrade tests only for the last month.
something else properly changed to surface it more often,

I think we need it figured before we branch out 5.4

**Comment by michoecho** [comment]
> this has started failing/coredumping half of our rolling upgrade tests only for the last month.
something else properly changed to surface it more often,

It's a timing-dependent bug. Maybe some patch last month just changed the timings slightly. It doesn't seem like something worth investigating.

**Comment by aleksbykov** [comment]
Issue reproduced in job:

## Installation details

Kernel Version: 4.18.0-513.el8.x86_64
Scylla version (or git commit hash): `5.3.0~rc0-20230523.3c3621db072d` with build-id `78cb9caaf1cdd0b37c3a38fa83548e6c4e3d4ab4`
Target Scylla version (or git commit hash): `5.4.0~dev-20230930.ecceb554c3a4` with build-id `14e795e196e61f6eacf2cfabbc2678bcac70593a`




Cluster size: 4 nodes (n2-highmem-16)

Scylla Nodes used in this run:
- rolling-upgrade--centos-8-db-node-2ec7b924-0-4 (35.243.249.33 | 10.142.15.227) (shards: 14)
- rolling-upgrade--centos-8-db-node-2ec7b924-0-3 (34.148.124.56 | 10.142.15.226) (shards: 14)
- rolling-upgrade--centos-8-db-node-2ec7b924-0-2 (34.138.184.147 | 10.142.15.222) (shards: 14)
- rolling-upgrade--centos-8-db-node-2ec7b924-0-1 (34.138.111.90 | 10.142.15.217) (shards: 14)


OS / Image: `https://www.googleapis.com/compute/v1/projects/centos-cloud/global/images/family/centos-stream-8` (gce: undefined_region)

Test: `rolling-upgrade-centos8-test`
Test id: `2ec7b924-d02e-4ecc-b7aa-b1dcd3724415`
Test name: `scylla-master/rolling-upgrade/rolling-upgrade-centos8-test`
Test config file(s):

- [rolling-upgrade.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/fa73cde2127567560e3cf64260f55b3275a86a24/test-cases/upgrades/rolling-upgrade.yaml)


<details>
<summary>
Logs and commands
</summary>


- Restore Monitor Stack command: `$ hydra investigate show-monitor 2ec7b924-d02e-4ecc-b7aa-b1dcd3724415`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=2ec7b924-d02e-4ecc-b7aa-b1dcd3724415)
- Show all stored logs command: `$ hydra investigate show-logs 2ec7b924-d02e-4ecc-b7aa-b1dcd3724415`


## Logs:
- **db-cluster-2ec7b924.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2ec7b924-d02e-4ecc-b7aa-b1dcd3724415/20230930_065741/db-cluster-2ec7b924.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2ec7b924-d02e-4ecc-b7aa-b1dcd3724415/20230930_065741/db-cluster-2ec7b924.tar.gz)
- **sct-runner-events-2ec7b924.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2ec7b924-d02e-4ecc-b7aa-b1dcd3724415/20230930_065741/sct-runner-events-2ec7b924.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2ec7b924-d02e-4ecc-b7aa-b1dcd3724415/20230930_065741/sct-runner-events-2ec7b924.tar.gz)
- **sct-2ec7b924.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2ec7b924-d02e-4ecc-b7aa-b1dcd3724415/20230930_065741/sct-2ec7b924.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2ec7b924-d02e-4ecc-b7aa-b1dcd3724415/20230930_065741/sct-2ec7b924.log.tar.gz)
- **loader-set-2ec7b924.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2ec7b924-d02e-4ecc-b7aa-b1dcd3724415/20230930_065741/loader-set-2ec7b924.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2ec7b924-d02e-4ecc-b7aa-b1dcd3724415/20230930_065741/loader-set-2ec7b924.tar.gz)
- **monitor-set-2ec7b924.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2ec7b924-d02e-4ecc-b7aa-b1dcd3724415/20230930_065741/monitor-set-2ec7b924.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2ec7b924-d02e-4ecc-b7aa-b1dcd3724415/20230930_065741/monitor-set-2ec7b924.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/rolling-upgrade/job/rolling-upgrade-centos8-test/217/)
[Argus](https://argus.scylladb.com/test/bc02ce5e-a796-4435-a3e2-c7a1744e3111/runs?additionalRuns[]=2ec7b924-d02e-4ecc-b7aa-b1dcd3724415)
</details>

**Comment by mykaul** [comment]
@eliransin - assigning it to you, assuming it's related/same as https://github.com/scylladb/scylladb/issues/14011

**Comment by mykaul** [comment]
@eliransin - ping

**Comment by mykaul** [comment]
This is waiting for a backport of https://github.com/scylladb/scylladb/issues/14011 to 5.2, which will then get it into 2023.1.(hopefully .4)

**Comment by eliransin** [comment]
> This is waiting for a backport of #14011 to 5.2, which will then get it into 2023.1.(hopefully .4)

I think this issue can be closed. The fix was backported to 5.4 and doesn't need to be backported to 5.2.
Here is the backport link:
https://github.com/scylladb/scylladb/commit/6d01d01debe6cb0cb52840620efd63bad98b2958

**Comment by eliransin** [comment]
This was fixed and it appears to have never been seen again since the aforementioned fix. closing.
